### ISSUE TITLE
【bug】unset  command will cause the variable unrecognized

### ISSUE THREAD

**Issue by mylovesaber** [BashSupport-Pro/bashsupport-pro#173]
demo script:
```shell
#!/bin/bash
aaa=1
bbb=${aaa}
echo $bbb  # output:  1

unset aaa
echo $bbb  # output:  1
```
Variable `aaa` comes from other ini file. In the main shell script, I need to replace variable name `aaa` by `bbb`. Then `aaa` doesn't need any more, I want to unset it.  

Since the value of `aaa` has been given to `bbb`, even I execute `unset aaa`, it doesn't affect the value of `bbb`.

Then I delete `echo $bbb` for all place, and add a new line with `echo `（echo and space）

```shell
#!/bin/bash
aaa=1
bbb=${aaa}
unset aaa
echo 
```

---

# Problem

I typed `$b`, BSP(Bashsupport Pro) should find and show the variable name `bbb` for me, However, it doesn't do this:
![image](https://github.com/user-attachments/assets/e20f68c3-044d-451e-9802-d94fe17af2a2)
Even I typed all the variable name `$bbb`, it still shows nothing.

If I delete `bb` or `bbb`(at least keep `$`, don't keep $bb, still not recognized), then click mouse on any other line in editor and click back to the line I would input `echo $bbb`.

 Input `b` and make the variable like `$bb`. BSP finally found variable name `$bbb`:
![image](https://github.com/user-attachments/assets/31025cb4-df57-4655-97e8-1916381c46cc)

BSP finally shows the whole variable only when  the variable  characters  length has two characters  longer that what I input.

Eg:
```shell
#!/bin/bash
aaa=1
bbbbbbbbbb=${aaa}
unset aaa
echo $bbbbbbbbbb
```

variable is `bbbbbbbbbb` (10 characters)
delete b until the variable like:
```bash
$b
$bb
$bbb
$bbbb
$bbbbb
$bbbbbb
$bbbbbbb
$bbbbbbbb
```
click mouse on any other line and click back, BSP recognize the final variable `$bbbbbbbbbb`

**Comment by mylovesaber** [comment]
this bug happens in many versions.

**Comment by jansorg** [comment]
Thank you for the report. This is indeed a bug and I'll fix this for the next update.
I think it's a problem of code completion, as operations like "go to reference" or "rename" still work on `$bbb`  after the `unset $aaa`.

**Comment by jansorg** [comment]
I've fixed this for 4.2.3, which will be releases soon

**Comment by jansorg** [comment]
Version 4.2.3 is available now with a fix
